oauth2_provider/models.py

```diff
@@ -342,11 +342,19 @@ def revoke(self):
         """
         Mark this refresh token revoked and revoke related access token
         """
-        access_token_model = get_access_token_model()
-        access_token_model.objects.get(id=self.access_token_id).revoke()
-        self.access_token = None
-        self.revoked = timezone.now()
-        self.save()
+        with transaction.atomic():
+            access_token_model = get_access_token_model()
+            refresh_token_model = get_refresh_token_model()
```

Comment:
Those first two lines don't need to be in transaction.atomic().

Reply:
Thanks, updated!